src/anchor/extractors/detect.py

```
@@ -0,0 +1,20 @@
+}
+
+
+def detect_format(path: str | Path) -> str:
+    """Return the logical format name for *path* based on its file extension."""
+    suffix = Path(path).suffix.lower()
+    try:
+        return _EXT_MAP[suffix]
+    except KeyError:
+        raise ValueError(
+            f"Unsupported file extension {suffix!r}: cannot detect format for {path!r}"
+        )
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)